linux_docker_resources/Dockerfile

```diff
@@ -175,13 +175,6 @@ RUN useradd -u 1234 -m rosbuild
 RUN sudo -H -u rosbuild -- git config --global user.email "jenkins@ci.ros2.org"
 RUN sudo -H -u rosbuild -- git config --global user.name "Jenkins ROS 2"
 RUN echo 'rosbuild ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
-WORKDIR "@workdir"
-
-# Add an entry point which changes rosbuild's UID from 1234 to the UID of the invoking user.
-# This means that the generated files will have the same ownership as the host OS user.
-ADD entry_point.sh /entry_point.sh
-RUN chmod 755 /entry_point.sh
```

Comment:
The changes here may be unnecessary strictly speaking but since we're now pushing these to the docker container using runtime arguments instead it makes sense to me to cut these layers out and avoid confusion. But I could also see arguments for maintaining "defaults" in the dockerfile. However since `@workdir` is a placeholder string that is filled in via sed at runtime, that's not a great default....

Reply:
I pulled this out and expanded on it in #641.